x-pack/plugins/infra/server/lib/host_details/process_list.ts

```diff
@@ -4,61 +4,134 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ProcessListAPIRequest, MetricsAPIRequest } from '../../../common/http_api';
-import { getAllMetricsData } from '../../utils/get_all_metrics_data';
-import { query } from '../metrics';
+import { ProcessListAPIRequest, ProcessListAPIQueryAggregation } from '../../../common/http_api';
 import { ESSearchClient } from '../metrics/types';
+import { CMDLINE_FIELD } from './common';
 
 export const getProcessList = async (
-  client: ESSearchClient,
-  { hostTerm, timerange, indexPattern }: ProcessListAPIRequest
+  search: ESSearchClient,
+  { hostTerm, timefield, indexPattern, to, sortBy, searchFilter }: ProcessListAPIRequest
 ) => {
-  const queryBody = {
-    timerange,
-    modules: ['system.cpu', 'system.memory'],
-    groupBy: ['system.process.cmdline'],
-    filters: [{ term: hostTerm }],
-    indexPattern,
-    limit: 9,
-    metrics: [
-      {
-        id: 'cpu',
-        aggregations: {
-          cpu: {
-            avg: {
-              field: 'system.process.cpu.total.norm.pct',
+  const body = {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              [timefield]: {
+                gte: to - 60 * 1000, // 1 minute
```

Comment:
Why only one minues? I'm seeing a difference between the total count of processes between this PR and the previous implementation. Wondering if this could be the reason.

Reply:
Primarily to optimize performance, but I think it makes sense given that the Inventory view presents a snapshot of the past 1 minute of data. Are you able to tell if the lower Total Processes count is because some of the processes had ended in the past 15 minutes, or if it's actually failing to pick up a bunch of still active processes?

Reply:
It's missing active processes 
<img width="736" alt="Screen Shot 2020-12-03 at 10 46 03 AM" src="https://user-images.githubusercontent.com/1482089/101067541-48877680-355d-11eb-8503-11a7f81024ea.png">
<img width="715" alt="Screen Shot 2020-12-03 at 10 43 46 AM" src="https://user-images.githubusercontent.com/1482089/101067548-4a513a00-355d-11eb-98a4-3f2707f18c6b.png">

Reply:
Should be fixed in the latest commit